### PR TITLE
Cdsupina/motion2d refactor

### PR DESCRIPTION
**Pull request by cdsupina**
General motion in Motion2DSystem

Enemy motion in EnemyMotion2DSystem.

### CODE REVIEW

File: src/systems/motion2d.rs [hunk 1]
```diff
@@ -1,35 +1,212 @@
...
+            // accelerate in negative y direction
+            motion_2d.velocity.y += motion_2d.acceleration.y;
+        }
+        EnemyType::Missile => {
+            // accelerate in negative y direction
+            motion_2d.velocity.y += motion_2d.acceleration.y;
+        }
+        EnemyType::RepeaterBody => {
+            // move down to position and then accelerate backwards
+            if transform.translation().y > ARENA_MIN_Y + ARENA_HEIGHT - 30.0 {
```
Comment (tigleym) [inline src/systems/motion2d.rs:139]: Is 30.0 an offset value of the enemy's hitbox? How are you getting this number?

File: src/systems/motion2d.rs [hunk 2]
```diff
@@ -1,35 +1,212 @@
...
+                motion_2d.velocity.y += motion_2d.acceleration.y;
+            } else {
+                motion_2d.velocity.y -= motion_2d.acceleration.y;
+            }
+        }
+        EnemyType::RepeaterHead => {
+            // move down to position and then accelerate backwards
+            if transform.translation().y > ARENA_MIN_Y + ARENA_HEIGHT - 67.0 {
```
Comment (tigleym) [inline src/systems/motion2d.rs:147]: Same as above.

File: src/systems/motion2d.rs [hunk 3]
```diff
@@ -1,35 +1,212 @@
...
+                motion_2d.velocity.y += motion_2d.acceleration.y;
+            } else {
+                motion_2d.velocity.y -= motion_2d.acceleration.y;
+            }
+        }
+        EnemyType::RepeaterShoulder => {
+            // move down to position and then accelerate backwards
+            if transform.translation().y > ARENA_MIN_Y + ARENA_HEIGHT - 32.0 {
```
Comment (tigleym) [inline src/systems/motion2d.rs:155]: same as above

**Review by tigleym — CHANGES_REQUESTED** [review]
Thanks! It's looking good so far. I just had some questions about the magic numbers being used for the enemy motion system.

File: src/systems/motion2d.rs (hunk 1 above)
Comment (cdsupina) [reply src/systems/motion2d.rs:139]: These were used in the previous enemy.rs system as well. These arbitrary numbers just indicate a position on the y axis in the arena. Once the parts reach the arbitrary position they stop accelerating downward and reverse acceleration in the other direction resulting in the oscillation. These arbitrary numbers line each of the boss parts up with each other to make it appear that the boss is one cohesive unit.

Comment (cdsupina) [reply src/systems/motion2d.rs:139]: I think we should leave this as is until we work on the boss part hierarchy.

**Review by tigleym — APPROVED** [review]
Thanks! LGTM.

**Pull request merged**